### PR TITLE
Kden branch xd

### PR DESCRIPTION
**Pull request by cadenforrest**
i have created a hero finder for ur review sir

### CODE REVIEW

File: secret.py [hunk 1]
```diff
@@ -0,0 +1 @@
+OPENDOTA_API_KEY = "5ab63e85-ff3a-4d29-a227-a78a362c42c4"
```
Comment (kmorrison) [inline secret.py]: we said never to commit this file ECKKKSS DEEEEE

File: tempCodeRunnerFile.py [hunk 2]
```diff
@@ -0,0 +1 @@
+find_juggernaut()
```
Comment (kmorrison) [inline tempCodeRunnerFile.py]: if you want to add runner, I often just add it in the main file

File: opendota.py [hunk 3]
```diff
@@ -1,11 +1,32 @@
...
+    if (herolist[i]['localized_name'] == heroname):
+      return(herolist[i])
+  print("Didn't find the hero")
+  return
+
+  
+def get_hero_list(): 
+  response = requests.get(
+    f'{API_ROOT}/heroes',
+    params = dict(api_key = secret.OPENDOTA_API_KEY)
```
Comment (kmorrison) [inline opendota.py:27]: in python args, we often don't include spaces, so like
`params=dict(api_key=secret.....`

File: opendota.py [hunk 4]
```diff
@@ -1,11 +1,32 @@
...
+
+def find_hero(heroname): 
+  i=1
+  herolist = get_hero_list()
+  for hero in herolist: 
+    i+=1
+    if (herolist[i]['localized_name'] == heroname):
+      return(herolist[i])
+  print("Didn't find the hero")
+  return
```
Comment (kmorrison) [inline opendota.py]: how about

```
for hero in get_hero_list():
    if hero['localized_name'] == heroname:
        return hero
print...
return
```

File: find_hero_test.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+herolist = opendota.get_hero_list()
+
+i = 1
+
+def find_hero(heroname): 
+  i=1
+  for hero in herolist: 
+    i+=1
+    if (herolist[i]['localized_name'] == heroname):
+      return(herolist[i])
```
Comment (kmorrison) [inline find_hero_test.py:12]: this looks very similar to other find_hero func. do we need it?

**Review by kmorrison — CHANGES_REQUESTED** [review]

File: find_hero_test.py (hunk 5 above)
Comment (cadenforrest) [reply find_hero_test.py:12]: nop

**Review by kmorrison — APPROVED** [review]

**Pull request merged**